libguard/guard_interface.cpp

```diff
@@ -124,6 +124,25 @@ static GuardRecord getHostEndiannessRecord(const GuardRecord& record)
     return convertedRecord;
 }
 
+/**
+ * @brief Function to get the size remaining in the file
+ *
+ * @param[in] size - File size
+ * @param[in] count - No of records present
+ *
+ */
+void fileSizeCheck(uint32_t size, uint32_t count)
+{
+    GuardRecord record;
+    uint32_t aval_size = size - (count * sizeof(record) + headerSize);
+
+    if (aval_size < sizeof(record))
+    {
+        guard_log(GUARD_ERROR, "Size left in file is %d\n", aval_size);
+        throw std::runtime_error("Enough size is not available in guard file");
```

Comment:
What exception you will throw in this case? The list which you shared is covered this case?

Reply:
Yes its covered.  guardMaxSizeReached